config/src/lib.rs

```diff
@@ -1187,39 +1187,205 @@ impl Config {
 
         dir_size_recursive(fs::read_dir(chain_path)?)
     }
+
+    fn fix_toml_for_profiles(toml_path: impl AsRef<Path>, profiles: &[Profile]) {
+        // parse toml
+        let toml_path = toml_path.as_ref();
+        let mut toml_doc: toml_edit::Document = match std::fs::read_to_string(toml_path) {
+            Ok(toml_str) => match toml_str.parse() {
+                Ok(value) => value,
+                Err(err) => {
+                    config_warn!("Invalid TOML at {}: {}", toml_path.display(), err);
+                    return
+                }
+            },
+            Err(err) => {
+                config_warn!("Could not read TOML at {}: {}", toml_path.display(), err);
+                return
+            }
+        };
+        // routine for inserting a [<profile>] into [profile.<profile>]
+        let insert_profile = |toml_map: &mut toml_edit::Table,
+                              profile_str: &str,
+                              value: toml_edit::Item| {
+            if toml_map.get(Config::PROFILE_SECTION).is_none() {
+                // insert profile section at the beginning of the map
+                let mut profile_section = toml_edit::Table::new();
+                profile_section.set_position(0);
+                profile_section.set_implicit(true);
+                toml_map.insert(Config::PROFILE_SECTION, toml_edit::Item::Table(profile_section));
+            }
+            // check the profile map for structure and existing keys
+            let profile_map = toml_map.get_mut(Config::PROFILE_SECTION).unwrap();
```

Comment:
can we use `expect("exists; qed")` here and add comment that we insert this if it's missing